app/src/main/java/de/westnordost/streetcomplete/quests/baby_changing_table/AddBabyChangingTable.kt

```diff
@@ -13,11 +13,14 @@ class AddBabyChangingTable : OsmFilterQuestType<Boolean>() {
     override val elementFilter = """
         nodes, ways with
         (
-          (
-            (amenity ~ restaurant|cafe|fuel|fast_food or shop ~ mall|department_store)
-            and toilets = yes
-          )
-          or amenity = toilets
+          // ask for all restaurants and toilets
+          amenity ~ restaurant|toilets
+          or
+          // ask for fuel stations or shops which have toilets
+          ((amenity ~ fuel or (shop and shop !~ no|vacant)) and toilets=yes)
+          or
+          //ask for cafés and fast_food which have toilets or seating
+          (amenity ~ cafe|fast_food and (toilets=yes or indoor_seating=yes or outdoor_seating=yes))
```

Comment:
Comments are not allowed in the element filter syntax. Also, some parentheses can be avoided because `and` has higher precedence than `or`. And the last condition can be simplified by using a regex in the key. In general, see the cheatsheet in [ElementFilterExpression.kt](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/elementfilter/ElementFilterExpression.kt). All these changes don't change your intended logic:

```suggestion
          amenity ~ restaurant|toilets
          or (amenity ~ fuel or (shop and shop !~ no|vacant)) and toilets = yes
          or amenity ~ cafe|fast_food and ~toilets|indoor_seating|outdoor_seating = yes
```

Reply:
Thanks for your quick feedback! Is it okay to leave the comments away? Or shall I put the comments above the filter? I even saw, that sometimes the Github Issue URL is put in as comment..

Reply:
Adding comments that explain what the element filter does is IMO not necessary, since the element filter syntax is quite readable. Adding the issue URL as a comment is sometimes helpful to understand the reasoning and/or link to the discussion that lead to a specific piece of code. That could be helpful here, but IMO it's not necessary here either, since `git blame` can always lead you to the commit that last changed a line of code, and thus lead you to the PR.